### PR TITLE
Add Amazon Linux 2 configs

### PR DESCRIPTION
**Pull request by haroldjin**
This PR adds amazon linux 2 config files with references to gpg keys provided by this [PR for distribution-gpg-keys](https://github.com/xsuchy/distribution-gpg-keys/pull/24/commits/3601e86a359892eea54e57d3e900199b9ee4ed5b).

### CODE REVIEW

**Comment by xsuchy** [comment]
Can you please use templates and put there the common parts. Please check e.g., the recent epel-8-x86_64 config.

File: mock-core-configs/etc/mock/amazonlinux-2-aarch64.cfg [hunk 1]
```diff
@@ -0,0 +1,25 @@
+config_opts['root'] = 'amzn2-aarch64'
```
Comment (xsuchy) [inline mock-core-configs/etc/mock/amazonlinux-2-aarch64.cfg]: We usually had two dashes in the name of the root. Following the patter $name-$version-$arch. I would like to see here 'amzn-2-aarch64'. This is not mandatory in mock itself. But I have seen some code based on top of mock, which assumes this schema.

Comment (haroldjin) [reply mock-core-configs/etc/mock/amazonlinux-2-aarch64.cfg]: @xsuchy, thanks for the feedback. I've made changes to to ensure $name$-version-$arch is enforced as well as using config template.

**Comment by praiskup** [comment]
Can you please squash the commits?

**Comment by praiskup** [comment]
mock throws error with this config:
```
$ mock -r amazonlinux-2-x86_64 --shell
  File "/etc/mock/amazonlinux-2-x86_64.cfg", line 1, in <module>
    include('/etc/mock/templates/amazonlinux-2.tpl')

  File "/etc/mock/templates/amazonlinux-2.tpl", line 1

    config_opts['chroot_setup_cmd'] = 'install @buildsys-build yum’

                                                                  ^

SyntaxError: EOL while scanning string literal

ERROR: Error in configuration
```

**Comment by haroldjin** [comment]
@praiskup, thanks for the comment. I have squashed the comment and fixed the apostrophe character.

**Comment by praiskup** [comment]
Mock fails for me right after
```
 # /usr/bin/yum --installroot /var/lib/mock/amzn-2-x86_64/root/ --releasever 2 install @buildsys-build yum --setopt=tsflags=nocontexts
```
If this is working fine _on amzn host_, I would say we could merge this in.  But it should be documented somewhere that it doesn't work elsewhere.

**Pull request merged**